Automattic/wp-calypso
# Hide post title in Gutenberg depending on theme mod.

### Changes proposed in this Pull Request
1. Add a "common" package to the FSE plugin for small bits of code that don't fit any of the main sub-features. This code is always executed.
2. Add the `hide-homepage-title` class to the body if editing the homepage in Gutenberg and if the theme mod is active.
3. Add style to not display the title in Gutenberg if the above class exists.

### FAQ:
- _Why not put this in wpcom-block-editor?_ That app isn't a WordPress plugin with php files that can be loaded wherever it is installed. Since we need to run php on simple and Atomic sites, the FSE is a good fit. I didn't want to have to update simple _and_ wpcomsh to support this on Atomic since it is harder to maintain. 
- _Why not put this in a different FSE sub-feature?_ It didn't seem to fit anything we had. Plus, it didn't make sense to add a "hide homepage title" subfeature since it's only like 10loc outside of the bootstrapping.
- _Why is this not in the theme codebase?_ We need to fix this issue for any theme which supports the homepage title mod, which includes 2019, 2020, and all Varia child themes. And probably some others. Unfortunately, none of those themes added support for hiding the title in the editor. If we wanted to do that, we'd have to add this identical code to _every single theme_ and then deploy them. That's a huge effort -- and since this is a Gutenberg classname, it's prone to get out of date. So to fix it, we'd have to then update all of those themes again. That's not worth it. Having this in a single place makes a ton of sense. Since the themes take care of rendering their own front end and homepages, they already support the front-end styles for this theme mod. Think of this code as "editor support" for that theme mod. Now, any theme that supports the mod will work correctly in the editor.
- _Doesn't Varia add this classname to the body already?_ Yes! However, it doesn't have css hiding the editor post title. In fact, this php code is copied from the Varia implementation.

### Testing instructions
1. Sync this to your sandbox.
2. Activate a Varia child theme.
3. Edit the homepage in Gutenberg. The post title should not be visible.
4. Activate the 2020 theme.
5. Edit the homepage in Gutenberg. The post title _should_ be visible. (It is visible since 2020 does not hide the homepage title by default until you change the setting.)
6. Go to the customizer, under "Homepage settings" and check "Hide homepage". Publish the changes.
7. Go back to the homepage editor. The post title should not be visible any more.
8. Make sure the homepage title is still hidden on the front end.
9. Make sure non-homepage post/page titles are still visible in the editor.


Fixes #37509, resolves #40410
